packages/asset-swapper/contracts/src/BancorSampler.sol

```diff
@@ -0,0 +1,131 @@
+    /// @return makerTokenAmounts Maker amounts bought at each taker token
+    ///         amount.
+    function sampleSellsFromBancor(
+        address[][] memory paths,
+        address takerToken,
+        address makerToken,
+        uint256[] memory takerTokenAmounts
+    )
+        public
+        view
+        returns (address bancorNetwork, address[] memory path, uint256[] memory makerTokenAmounts)
+    {
+        bancorNetwork = _getBancorNetwork();
+        if (paths.length == 0) {
+            return (bancorNetwork, path, makerTokenAmounts);
+        }
+        uint256 maxBoughtAmount = 0;
+        // Find the best path by selling the largest taker amount
+        for (uint256 i = 0; i < paths.length; i++) {
+            if (paths[i].length < 2) {
+                continue;
+            }
+
+            try
+                IBancorNetwork(bancorNetwork)
+                    .rateByPath
+                        {gas: BANCOR_CALL_GAS}
+                        (paths[i], takerTokenAmounts[takerTokenAmounts.length-1])
+                returns (uint256 amount)
+            {
+                if (amount > maxBoughtAmount) {
+                    maxBoughtAmount = amount;
+                    path = paths[i];
+                }
+            } catch (bytes memory) {
+                // Swallow failures, leaving all results as zero.
+                continue;
+            }
+        }
+
+        uint256 numSamples = takerTokenAmounts.length;
+        makerTokenAmounts = new uint256[](numSamples);
+
+        for (uint256 i = 0; i < numSamples; i++) {
+            try
+                IBancorNetwork(bancorNetwork)
+                    .rateByPath
+                        {gas: BANCOR_CALL_GAS}
+                        (path, takerTokenAmounts[i])
+                returns (uint256 amount)
+            {
+                makerTokenAmounts[i] = amount;
+            } catch (bytes memory) {
+                // Swallow failures, leaving all results as zero.
+                break;
+            }
+        }
+        return (bancorNetwork, path, makerTokenAmounts);
+    }
+
+    /// @dev Sample buy quotes from Bancor. Unimplemented
+    /// @param paths The paths to check for Bancor. Only the best is used
+    /// @param takerToken Address of the taker token (what to sell).
+    /// @param makerToken Address of the maker token (what to buy).
+    /// @param makerTokenAmounts Maker token buy amount for each sample.
+    /// @return bancorNetwork the Bancor Network address
+    /// @return path the selected conversion path from bancor
+    /// @return takerTokenAmounts Taker amounts sold at each maker token
+    ///         amount.
+    function sampleBuysFromBancor(
+        address[][] memory paths,
+        address takerToken,
+        address makerToken,
+        uint256[] memory makerTokenAmounts
+    )
+        public
+        view
+        returns (address bancorNetwork, address[] memory path, uint256[] memory takerTokenAmounts)
+    {
```

Comment:
What's the plan for this?

Reply:
Figure it out one day. Will probably have to do what we do with Curve here.